app/src/test/java/com/concordia/mcga/lib/BuildingBottomSheetInfoTest.java

```diff
@@ -0,0 +1,175 @@
+        Parcelable mState = bottomSheet.onSaveInstanceState(fakeCoordinatorLayout, fakeChildView);
+        int state2 = bottomSheet.getState();
+
+        assertTrue(state == state2);
+
+        bottomSheet.onRestoreInstanceState(fakeCoordinatorLayout, fakeChildView, mState);
+
+        int state3 = bottomSheet.getmLastStableState();
+
+        assertTrue(state2 == state3);
+
+        bottomSheet.setState(BuildingBottomSheetInfo.STATE_HIDDEN);
+        state2 = bottomSheet.getState();
+        state3 = bottomSheet.getmLastStableState();
+        assertTrue(state2 == state3);
+    }
+
+    @Test
+    public void CoordinatorLayout_ShouldNotBeNull_True(){
+        assertNotNull(fakeCoordinatorLayout);
+    }
+
+    @Test
+    public void onLayoutchild_ShouldNotBeNull_True(){
+        assertTrue(bottomSheet.onLayoutChild(fakeCoordinatorLayout, fakeChildView, 0));
+    }
+
+
+    @Test
+    public void touchEvent_isDetected_False(){
+        MotionEvent event = MotionEvent.obtain(200, 300, MotionEvent.ACTION_DOWN, 15.0f, 10.0f, 0);
+        fakeChildView.setVisibility(View.INVISIBLE);
+        event = MotionEvent.obtain(200, 300, MotionEvent.ACTION_UP, 15.0f, 10.0f, 0);
+        assertFalse(bottomSheet.onTouchEvent(fakeCoordinatorLayout, fakeChildView, event));
+
+    }
+
+    @Test
+    public void scrollVelocity_isUpdatedAfterEvent_True(){
+        int[] consumed = {0, 300};
+        bottomSheet.onNestedPreScroll(fakeCoordinatorLayout, fakeChildView, view, 0, 0, consumed);
+        assertTrue(bottomSheet.getmNestedScrolled());
+    }
+
+    @Test
+    public void mNestedScroll_isFalseWhenNotScrolling_True(){
+        bottomSheet.onStopNestedScroll(fakeCoordinatorLayout, fakeChildView, view);
+        assertFalse(bottomSheet.getmNestedScrolled());
+    }
+
+    @Test
+    public void setPeekHeight_updatesHeightAndOffset_true(){
+        bottomSheet.setPeekHeight(700);
+        assertTrue(bottomSheet.getPeekHeight() == 700);
+        assertTrue(bottomSheet.getmMaxOffset() == fakeCoordinatorLayout.getHeight() - 700);
+    }
+
+    @Test
+    public void addBottomSheetCallBack_UpdateBottomSheetCallBack_True(){
+        bottomSheet.addBottomSheetCallback(null);
+        Vector<BuildingBottomSheetInfo.BottomSheetCallback> vector = bottomSheet.getmCallBack();
+        assertTrue(vector.get(0) == null);
+
+        bottomSheet.addBottomSheetCallback(new BuildingBottomSheetInfo.BottomSheetCallback() {
+            @Override
+            public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                switch (newState) {
+                    case BuildingBottomSheetInfo.STATE_COLLAPSED:
+                        break;
+
+                    case BuildingBottomSheetInfo.STATE_EXPANDED:
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+            @Override
+            public void onSlide(@NonNull View bottomSheet, float slideOffset) {
```

Comment:
. It would look less suspicious if there was that comment inside the method. Instead of having a developer wonder in the future why this method is overriden and empty.